mlir/python/PyCompilerDriver.cpp

```diff
@@ -77,13 +97,13 @@ PYBIND11_MODULE(compiler_driver, m)
     m.def(
         "run_compiler_driver",
         [](const char *source, const char *workspace, const char *moduleName, bool keepIntermediate,
-           bool verbose, py::list pipelines,
-           bool lower_to_llvm) -> std::unique_ptr<CompilerOutput> {
+           bool verbose, py::list pipelines, bool lower_to_llvm,
+           bool multi_threaded_compilation) -> std::unique_ptr<CompilerOutput> {
             // Install signal handler to catch user interrupts (e.g. CTRL-C).
             signal(SIGINT,
                    [](int code) { throw std::runtime_error("KeyboardInterrupt (SIGINT)"); });
 
-            std::unique_ptr<CompilerOutput> output(new CompilerOutput());
+            std::unique_ptr<CompilerOutput> output(new CompilerOutput(initDumpCounter(workspace)));
```

Comment:
I see, here is where you create it. How about instead of having the `initDumpCounter` function, just have `run_compiler_driver` take an int pararameter which defaults to 1. And have the canonicalizer set it to 0.

I would like to avoid having the `initDumpCounter` function.

Reply:
@erick-xanadu in this case we will would need to write a Python analog of `initDumpCounter` counting the canonicalizer's artifacts and setting the integer to the right value. I'm ok with this solution, but do you think it is worth the time to implement?

Reply:
The canonicalizer will always have 1 single output. I don't think we need to count anything.

Reply:
Why not? What if someone change it?